x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/common/H3CartesianUtil.java

```diff
@@ -51,24 +44,24 @@ public final class H3CartesianUtil {
             }
         }
         for (int res = 2; res <= H3.MAX_H3_RES; res++) {
-            CACHED_H3.put(NORTH[res], getCoordinates(NORTH[res]));
-            CACHED_H3.put(SOUTH[res], getCoordinates(SOUTH[res]));
+            CACHED_H3.put(H3.northPolarH3(res), getCoordinates(H3.northPolarH3(res)));
+            CACHED_H3.put(H3.southPolarH3(res), getCoordinates(H3.southPolarH3(res)));
         }
     }
 
     private static final double[] NORTH_BOUND = new double[16];
     private static final double[] SOUTH_BOUND = new double[16];
     static {
         for (int res = 0; res <= H3.MAX_H3_RES; res++) {
-            NORTH_BOUND[res] = toBoundingBox(NORTH[res]).getMinY();
-            SOUTH_BOUND[res] = toBoundingBox(SOUTH[res]).getMaxY();
+            NORTH_BOUND[res] = toBoundingBox(H3.northPolarH3(res)).getMinY();
+            SOUTH_BOUND[res] = toBoundingBox(H3.southPolarH3(res)).getMaxY();
         }
     }
 
     /** For the given resolution, it returns the maximum latitude of the h3 bin containing the south pole */
```

Comment:
This comment does not match the behaviour of the method, and seems to have been copied from another method.

Reply:
fixed the comment